assets/js/src/tracker/index.js

```diff
@@ -0,0 +1,104 @@
+import { config } from '../config';
+
+/**
+ * A tracking utility for initializing a GA4, managing accepted
+ * events, attaching data to those events, and tracking them.
+ *
+ * @param {Object} config Configuration settings for the tracker.
+ *
+ * @return {Object} An object containing methods: init, setupEvents, event{ attach, get, track }.
+ */
+export const tracker = ( () => {
+	const events = [];
+
+	return {
+		/**
+		 * Initializes the tracker and dataLayer if not already done.
+		 */
+		init: () => {
+			if ( window[ config.tracker_var ] ) {
+				// Tracker already initialized. Do nothing.
+				return;
+			}
+
+			window.dataLayer = window.dataLayer || [];
+			window[ config.tracker_var ] = function () {
+				window.dataLayer.push( arguments );
+			};
+			window[ config.tracker_var ]( 'js', new Date() );
+			window[ config.tracker_var ](
+				'set',
+				`developer_id.${ config.developer_id }`,
+				true
+			);
+			window[ config.tracker_var ]( 'config', config.gtag_id, {
+				allow_google_signals: config.allow_google_signals,
+				link_attribution: config.link_attribution,
+				anonymize_ip: config.anonymize_ip,
+				logged_in: config.logged_in,
+				linker: config.linker,
+				custom_map: config.custom_map,
+			} );
```

Comment:
:nail_care: WDYT of:?

```suggestion
			function gtag() {
				window.dataLayer.push( arguments );
			};
			window[ config.tracker_var ] = gtag;
			gtag( 'js', new Date() );
			gtag(
				'set',
				`developer_id.${ config.developer_id }`,
				true
			);
			gtag( 'config', config.gtag_id, {
				allow_google_signals: config.allow_google_signals,
				link_attribution: config.link_attribution,
				anonymize_ip: config.anonymize_ip,
				logged_in: config.logged_in,
				linker: config.linker,
				custom_map: config.custom_map,
			} );
```

Reply:
Or if we really want to keep the `window.dataLayer[0].callee.name === ''` (correct me if I'm wrong, but I don't think we have such a need)
```js
const gtag = window[ config.tracker_var ] = function() {
```